File: app/src/main/java/com/lumiyaviewer/lumiya/data/repository/SearchGridResultRepositoryAdapter.java
package com.lumiyaviewer.lumiya.data.repository;

import androidx.annotation.Nullable;
import com.lumiyaviewer.lumiya.dao.SearchGridResult;
import com.lumiyaviewer.lumiya.dao.SearchGridResultDao;
import com.lumiyaviewer.lumiya.data.room.SearchGridResultEntity;
import com.lumiyaviewer.lumiya.data.room.SearchGridResultRoomDao;

public class SearchGridResultRepositoryAdapter {
    private final SearchGridResultDao greenDao;
    @Nullable
    private final SearchGridResultRoomDao roomDao;

    public SearchGridResultRepositoryAdapter(SearchGridResultDao greenDao, @Nullable SearchGridResultRoomDao roomDao) {
        this.greenDao = greenDao;
        this.roomDao = roomDao;
    }

    public void insert(SearchGridResult value) {
        this.greenDao.insert(value);
        if (this.roomDao != null) {
            this.roomDao.insert(toEntity(value));
        }
    }

    public void deleteOtherQueries(java.util.UUID searchUUID) {
        this.greenDao.queryBuilder().where(SearchGridResultDao.Properties.SearchUUID.notEq(searchUUID), new de.greenrobot.dao.query.WhereCondition[0]).buildDelete().executeDeleteWithoutDetachingEntities();
        if (this.roomDao != null) {
            this.roomDao.deleteOtherQueries(searchUUID);
        }
    }

    private static SearchGridResultEntity toEntity(SearchGridResult value) {
        SearchGridResultEntity entity = new SearchGridResultEntity();
        entity.id = value.getId();
        entity.searchUUID = value.getSearchUUID();
        entity.itemType = value.getItemType();
        entity.itemUUID = value.getItemUUID();
        entity.itemName = value.getItemName();
        entity.levensteinDistance = value.getLevensteinDistance();
        entity.memberCount = value.getMemberCount();
        return entity;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/data/repository/SearchGridResultRepositoryAdapter.kt
package com.lumiyaviewer.lumiya.data.repository

import com.lumiyaviewer.lumiya.dao.SearchGridResult
import com.lumiyaviewer.lumiya.dao.SearchGridResultDao
import com.lumiyaviewer.lumiya.data.room.SearchGridResultEntity
import com.lumiyaviewer.lumiya.data.room.SearchGridResultRoomDao
import java.util.UUID

class SearchGridResultRepositoryAdapter(
    private val greenDao: SearchGridResultDao,
    private val roomDao: SearchGridResultRoomDao?
) {
    fun insert(value: SearchGridResult) {
        greenDao.insert(value)
        roomDao?.insert(value.toRoomEntity())
    }

    fun deleteOtherQueries(searchUUID: UUID) {
        greenDao.queryBuilder()
            .where(SearchGridResultDao.Properties.SearchUUID.notEq(searchUUID))
            .buildDelete()
            .executeDeleteWithoutDetachingEntities()
        roomDao?.deleteOtherQueries(searchUUID)
    }

    private fun SearchGridResult.toRoomEntity() = SearchGridResultEntity().also { entity ->
        entity.id = id
        entity.searchUUID = searchUUID
        entity.itemType = itemType
        entity.itemUUID = itemUUID
        entity.itemName = itemName
        entity.levensteinDistance = levensteinDistance
        entity.memberCount = memberCount
    }
}
